### PR TITLE
docs(root): added code examples for ic-toast

### PR DESCRIPTION
**Pull request by jd239**
## Summary of the changes

Added component details for Toasts on the Toasts guidance page. 
This includes:
- Interactive example on guidance.
- Multiple interactive examples and code snippets in code.
- Updated guidance imagery

The build is currently failing due to the component not being available in `beta.8`.

## Related issue

#335 

## Checklist

- [ ] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: audit-ci.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "GHSA-w5p7-h5w8-2hfq",
     "GHSA-3rfm-jhwj-7488",
     "GHSA-hhq3-ff78-jv3g",
-    "GHSA-76p3-8jx3-jpfq"
+    "GHSA-76p3-8jx3-jpfq",
+    "GHSA-hc6q-2mpp-qw7j"
```
Comment (GCHQ-Developer-112) [inline audit-ci.json:12]: Can this vulnerability be fixed with `npm audit:fix`

Comment (jd239) [reply audit-ci.json:12]: I didn't realise at the time but it gets fixed by this PR: https://github.com/mi6/ic-design-system/pull/341. Will remove this entry from the whitelist

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-112 — DISMISSED** [review]

**Review by GCHQ-Developer-530 — DISMISSED** [review]

File: src/content/structured/components/toasts/code.mdx [hunk 2]
```diff
@@ -19,12 +19,466 @@ tabs:
...
+      </IcToastRegion>
+    </>
+  );
+}
+
+<ComponentPreview snippets={withVariantSnippet}>
+  <WithVariantToast />
+</ComponentPreview>
+
+### Multiline
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/toasts/code.mdx:161]:
```suggestion
### Multi-line
```

File: src/content/structured/components/toasts/code.mdx [hunk 3]
```diff
@@ -19,12 +19,466 @@ tabs:
...
+  {
+    language: "React",
+    snippet: `const toastRegionEl = useRef(null);
+const toastEl = useRef(null);
+const handleClick = async () => {
+  await toastRegionEl.current.setVisible(toastEl.current);
+}
+return (
+  <>
+    <IcButton onClick={handleClick}>Display multiline toast</IcButton>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/toasts/code.mdx:187]:
```suggestion
    <IcButton onClick={handleClick}>Display multi-line toast</IcButton>
```

File: src/content/structured/components/toasts/code.mdx [hunk 4]
```diff
@@ -19,12 +19,466 @@ tabs:
...
+
+export const MultilineToast = () => {
+  const toastRegionEl = useRef(null);
+  const toastEl = useRef(null);
+  const handleClick = async () => {
+    await toastRegionEl.current.setVisible(toastEl.current);
+  };
+  return (
+    <>
+      <IcButton onClick={handleClick}>Display multiline toast</IcButton>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/toasts/code.mdx:208]:
```suggestion
      <IcButton onClick={handleClick}>Display multi-line toast</IcButton>
```

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by DBD324 — APPROVED** [review]
Looks good. Thanks for the update.

**Review by GCHQ-Developer-112 — DISMISSED** [review]

File: src/content/structured/components/toasts/code.mdx [hunk 5]
```diff
@@ -19,12 +19,466 @@ tabs:
...
+      </IcToastRegion>
+    </>
+  );
+}
+
+<ComponentPreview snippets={slottedActionsSnippet}>
+  <SlottedActionsToast />
+</ComponentPreview>
+
+### Auto dismiss
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/toasts/code.mdx:341]:
```suggestion
### Auto-dismiss
```

File: src/content/structured/components/toasts/code.mdx [hunk 6]
```diff
@@ -19,12 +19,466 @@ tabs:
...
+
+export const AutoDismissToast = () => {
+  const toastRegionEl = useRef(null);
+  const toastEl = useRef(null);
+  const handleClick = () => {
+    toastRegionEl.current.setVisible(toastEl.current);
+  };
+  return (
+    <>
+      <IcButton onClick={handleClick}>Display auto dismiss toast</IcButton>
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/toasts/code.mdx:384]:
```suggestion
      <IcButton onClick={handleClick}>Display auto-dismiss toast</IcButton>
```

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**